strawberry/union.py

```diff
@@ -63,6 +64,35 @@ def __hash__(self) -> int:
         # TODO: Is this a bad idea? __eq__ objects are supposed to have the same hash
         return id(self)
 
+    def __or__(self, other: Union[StrawberryType, type]) -> StrawberryType:
+        if hasattr(other, "_type_definition"):
+            return StrawberryUnion(
+                type_annotations=(
+                    *self.type_annotations,
+                    StrawberryAnnotation(other),
+                )
+            )
```

Comment:
Maybe we can make a factory method for this? `StrawberryUnion.extend` or something?